src/pages/About/About.tsx

```diff
@@ -0,0 +1,283 @@
+import "./About.css";
+import {
+  handleLogin,
+  clientId,
+  scope,
+  redirectUrl,
+} from "../../utils/LoginFunctions";
+import {
+  dashOne,
+  dashTwo,
+  dashThree,
+  dashFour,
+  aboutStrava,
+  dashTable,
+  dashMap,
+  dashGraph,
+} from "../../assets/about";
+import { motion, useAnimation } from "framer-motion";
+import { useInView } from "react-intersection-observer";
+import { useEffect } from "react";
+import {
+  framerImageVariants,
+  framerTextVariants,
+  secondBlockChildren,
+  secondBlockVariants,
+} from "./utils/FramerVaraints";
+import { handleHover, handleHoverEnd } from "./utils/Functions";
+import { useShowAndHideAnimation } from "../../utils/customHooks";
+
+const About = () => {
+  const { ref: blockOne, inView: blockOneInView } = useInView({
+    threshold: 0.2,
+  });
+  const { ref: blockTwo, inView: blockTwoInView } = useInView({
+    threshold: 0.2,
+  });
+
+  const { ref: blockThree, inView: blockThreeInView } = useInView({
+    threshold: 0.3,
+  });
+  const { ref: blockFour, inView: blockFourInView } = useInView({
+    threshold: 0.3,
+  });
+  const { ref: blockFive, inView: blockFiveInView } = useInView({
+    threshold: 0.3,
+  });
+
+  const animation = useAnimation();
+  const textAnimation = useAnimation();
+  const animationTwo = useAnimation();
+  const textAnimationTwo = useAnimation();
+  const animationThree = useAnimation();
+  const textAnimationThree = useAnimation();
+  const animationFour = useAnimation();
+  const textAnimationFour = useAnimation();
+  const animationFive = useAnimation();
+  const textAnimationFive = useAnimation();
+
+  useEffect(() => {
+    if (blockOneInView) {
+      animation.start("visible");
+      textAnimation.start("visible");
+    }
+    if (!blockOneInView) {
+      animation.start("hidden");
+      textAnimation.start("hidden");
+    }
+  }, [blockOneInView]);
+
+  useShowAndHideAnimation(blockOneInView, animation, textAnimation);
+  useShowAndHideAnimation(blockTwoInView, animationTwo, textAnimationTwo);
+  useShowAndHideAnimation(blockThreeInView, animationThree, textAnimationThree);
+  useShowAndHideAnimation(blockFourInView, animationFour, textAnimationFour);
+  useShowAndHideAnimation(blockFiveInView, animationFive, textAnimationFive);
```

Comment:
When we have parts that are repeating we can extract the code and either create a reusable component, a function or a custom hook that can save us some lines of code :)

Try to keep the principle of DRY as much as possible.

Reply:
This is great. Thanks man!